src/lib/users/encryption_method.rb

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2016 SUSE LLC.
+#  All Rights Reserved.
+
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of version 2 or 3 of the GNU General
+#  Public License as published by the Free Software Foundation.
+
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+#  GNU General Public License for more details.
+
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, contact SUSE LLC.
+
+#  To contact SUSE about this file by physical or electronic mail,
+#  you may find current contact information at www.suse.com
+
+require "yast"
+Yast.import "UsersSimple"
+
+module Users
+  # Object oriented wrapper to read and set the password encryption method at
+  # Yast::UsersSimple
+  #
+  # @example
+  #   md5 = EncryptionMethod.new("md5") # Raises exception on wrong id
+  #   puts "The system is configured to use #{md5.label}" if md5.current?
+  class EncryptionMethod
+    class Error < RuntimeError
+    end
+
+    class NotFoundError < Error
+    end
+
+    extend Yast::I18n
+    textdomain "users"
+
+    attr_accessor :id
+
+    # "blowfish" is also known to UsersSimple, but it's not longer offered
+    # as a valid option. See fate#312321
+    LABELS = {
+      # TRANSLATORS: encryption type
+      "des"    => _("DES"),
+      # TRANSLATORS: encryption type
+      "md5"    => _("MD5"),
+      # TRANSLATORS: encryption type
+      "sha256" => _("SHA-256"),
+      # TRANSLATORS: encryption type
+      "sha512" => _("SHA-512")
+    }
+    private_constant :LABELS
+
+    DEFAULT_ID = "sha512"
+    private_constant :DEFAULT_ID
+
+    def initialize(id)
+      EncryptionMethod.raise_if_not_found(id)
```

Comment:
I think `raise_if_not_found` do not looks like nice method name. What about `validate!(id)`